### DISCUSSION TITLE
Update set picker highlight

### DISCUSSION THREAD

**Discussion by Zorakie** [arnoudkooi/SN-Utils#366] [Ideas]
So imagine this:
You do some development as part of a bigger project. Several developers working in parallel, multitude of update sets open not only for features being programmed, but also being tested. Long story short, when you open your update set picker, you have nice let's say 20+ rows.

But you do care only about your 3 update sets.

Now let's imagine, your project has some naming convention, where, for example, the name of the update set has not only like scope, sprint/release number and story number, but, for example, your name shortcut.

So what about SN utils would have some highlighting?
What if you would be able to define a part of string (contains "XYZ") and whatever name matches, would be highlighted?
Wouldn't it be great to see for the first glance, which 3 out of those 20+ update sets are yours?

2 ways to do it:
Simple:
In SN utils, comma separated input configurable per instance (so like 2 column table input). SN Utils will match every update set against this list, and if anything matches *item*, highlight it by one predefined color. 

More complex:
List configurable per instance, but every substring as separate input with its own order and configurable color. SN utils again tries to match every Update set name against this list, highlighting it with the color of the highest order match. Or even more complex, the row/option itself would not be highlighted, but there would be some colored dots of all matches on the side, or something like this.

In my opinion, the first solution is perfectly sufficient though.

I already do use this my own script through TamperMonkey to spot update sets with my name code easily (so basically what contains my name code, highlight whole option yellow), but it is not really compatible with the extended picker from SN Utils, so I would love to see some feature like this included.

Thanks!

**Comment by arnoudkooi** [comment]
Have you considered a slash command? have a look at the built in `/su` one, you could copy or modify that one

**Reply by Zorakie** [reply]
Well, it kind of does the trick, but from the UX perspective, it is very inconvenient.

So lets imagine this:

STRY8541651_XYZ
STRY8687651_ABC
STRY8541327_ZOR
STRY8557842_CDE
STRY8536587_XYZ
STRY8532178_ABC
STRY8532785_ABC
STRY8532785_ZOR
STRY8547895_XYZ
...
...

You want to switch from STRY8541327_ZOR to STRY8532785_ZOR.

What I am looking for is to click on Update set picker, and see my _ZOR update sets highlighted, so the switch takes literally 2 seconds as you just don't need to read through the list, where your update sets are, but you literally just focus on the color.

Your /su command is a bit helpful, but the execution takes time again: you need to write the command, then you need to write ZOR, then you need to read the list and then you need to smash the number, which one to select. Multiple step unnecessary process with writing every time you need to switch. Even if I consider command scrolling, still it is quite keyboard mushing, what you would be able to do here with color focus and 2 clicks

**Reply by arnoudkooi** [reply]
I see, but this would need to be done for UI16 as well as Next Experience.
Besides, I see a long list of features with minimal adoption, so I really want to push what can be done with the current feature set.

Please give this a try:

- Start by searching and editing the OOB `/su `command and make the below changes.
- Change field values:
- URL: `sys_update_set_list.do?sysparm_query=state=in progress^application=javascript:gs.getCurrentApplicationId()^nameENDSWITH_ZOR^nameLIKE$0^ORDERBYDESCsys_updated_on`
- Shortcut: zor (or whatever you like)
- Order: 1 (this will ensure this command always is displayed on top)
- Hint: adjust to your liking

![Google Chrome20230222 - 2359](https://user-images.githubusercontent.com/840781/220782574-de14c64c-b992-4ae5-a37e-90886c7aa5b1.png)

Advanced bonus option:
This will allow you to execute the command with the assigned keyboard shortcut
Update the URL of command `/shortcut` to `//zor` (Note the double slash)
In your extension shortcuts (chrome://extensions/shortcuts), make sure you assign a keyboard shortcut to this: 
![Google Chrome20230223 - 0009](https://user-images.githubusercontent.com/840781/220784400-dc5b95dc-51f9-472f-b8f4-6be0ffd739fe.png)
